code/game/machinery/fusion_engine.dm

```diff
@@ -232,23 +268,67 @@
 				fusion_cell = null
 				update_icon()
 				return TRUE
+
+	else if(HAS_TRAIT(O, TRAIT_TOOL_MULTITOOL))
+		if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_ENGI))
+			to_chat(user, SPAN_WARNING("You have no idea what to do with [src]."))
+			return
+
+		if(!overloaded)
+			if(!SShijack.sd_unlocked)
+				to_chat(user, SPAN_WARNING("You consider overloading [src]'s safeties, but you decide against it."))
+				return
+
+			if(inoperable())
+				to_chat(user, SPAN_WARNING("[src] needs to be working and have external power in order to overload it!"))
+				return
+
+			to_chat(user, SPAN_WARNING("You start overloading the safeties on [src]..."))
+			if(!do_after(user, 1.5 SECONDS, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+				return
+
+			to_chat(user, SPAN_WARNING("You finish overloading the safeties on [src]."))
+			set_overloading(TRUE)
+			log_game("[key_name(user)] has overloaded a generator.")
+
+		else
+			to_chat(user, SPAN_WARNING("You start restoring the safeties on [src]..."))
+			if(!do_after(user, 1.5 SECONDS, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+				return
+
+			to_chat(user, SPAN_WARNING("You finish restoring the safeties on [src]."))
+			log_game("[key_name(user)] has restored the safeties of a generator.")
+			set_overloading(FALSE)
```

Comment:
this needs an inoperable() check after the do_after too

Reply:
done